### PR TITLE
Add auto-start support

### PR DESCRIPTION
**Pull request by seblemaguer**
I am currently working on adding the auto-start support.

- [X] add the logic 
- [x] Add the documentation
- [x] Add the test

This aims to solve #38

### CODE REVIEW

**Comment by seblemaguer** [comment]
I took the time (sorry for that, I had problem to understand ERT) but now it should be ready to be merged if you want :)

**Comment by Fuco1** [comment]
Hey, looks pretty OK.  Some of the formatting could use some love :)

I'm not sure how I feel about the explicit call to the auto start method.

I think the idea was to have it auto-start immeiately after `prodigy-define-service` is finished.  However this might cause quite some trouble in case there are dependencies (which we don't have yet).  

What should happen if I call the auto start function and then define another auto-start process during my session? We should at least explain the expected behaviour.

**Comment by Fuco1** [comment]
Maybe we could use `after-init-hook` to call that function automatically?

**Comment by seblemaguer** [comment]
I have a look at this (about the formatting too :D)  and I update the pull request

**Comment by seblemaguer** [comment]
Hello, 

I had a little bit of time (lately) to rethink about it:
  - for the formatting, I actually don't really understand what is wrong. If you can tell me, that would be great as I can learn :)
  - for the method I added it in order to have a better control. The use `after-init-hook` seems a good idea for me.
  - for the testing, yes. I will see to add that.

File: prodigy.el [hunk 1]
```diff
@@ -1461,6 +1473,31 @@ beginning of the line."
...
   (use-local-map prodigy-view-mode-map))
 
+
+;;;###autoload
+(defun prodigy-auto-start ()
+  "Start services whose property `auto-start' is set to `t'."
+  (interactive)
+  (let ((saved-buffer (current-buffer))
+	(buffer-p (prodigy-buffer))
+        (buffer (get-buffer-create prodigy-buffer-name)))
```
Comment (Fuco1) [inline prodigy.el]: @seblemaguer 	For example here.  This should be aligned.  The `let` form body should be 2 spaces indented.

Use `indent-region` on the defun to use the default emacs lisp formatting rules, it is always correct.

Comment (seblemaguer) [reply prodigy.el]: ok thanks a lot :)

I update the pull request and tell you when I finished with it.

File: test/prodigy-process-handling-test.el [hunk 2]
```diff
@@ -34,22 +34,22 @@
 ;;;; prodigy-start-service
 
 (ert-deftest-async prodigy-start-service-test/callback-when-started (done)
-  (with-sandbox
-   (let ((service (prodigy-test/make-service)))
-     (prodigy-start-service service
-       (lambda ()
-         (prodigy-stop-service service nil done))))))
+                   (with-sandbox
```
Comment (Fuco1) [inline test/prodigy-process-handling-test.el]: `ert-deftest-async` is a macro defined in some helper file.  You need to load that file before reindenting as that will also load the macro's indent rule.  This is most certainly not correct.

**Comment by seblemaguer** [comment]
I refactor some stuff:
  - the indentation based on your comments (thanks by the way I was not away of the fact that you needed to load the helper file to get the indentation rules for the macro)
  - I also took into account your comment about the function. Now it doesn't depends if we activate the auto-start before or after. I didn't add the after-init-hook documentation but I can do that.
  - I rebase everything to be in 1 commit and to take into account the evolution of the master branch.

**Comment by seblemaguer** [comment]
I refactor some stuff:
  - the indentation based on your comments (thanks by the way I was not away of the fact that you needed to load the helper file to get the indentation rules for the macro)
  - I also took into account your comment about the function. Now it doesn't depends if we activate the auto-start before or after. I didn't add the after-init-hook documentation but I can do that.
  - I rebase everything to be in 1 commit and to take into account the evolution of the master branch.

**Comment by Fuco1** [comment]
@seblemaguer 	Thanks, looks good!  I'll try to review this ASAP but please if I forget don't be afraid of being annoying and just ask for review!  That helps me a lot to keep track of everything.

**Comment by seblemaguer** [comment]
@Fuco1 can we see to finish this pr? Else I am going to loose track of it for months again :/

**Comment by rejeep** [comment]
Please don't have inline comments, better describing them with a test. Comments get out of date.

**Comment by seblemaguer** [comment]
@rejeep I removed the comments

**Comment by seblemaguer** [comment]
Hello,

is there any news for this?

**Comment by DamienCassou** [comment]
I'm closing all old PRs. If you still care, please rebase your branch and reopen the PR.

**Comment by seblemaguer** [comment]
With context.el this pr became obsolete. Sorry I forgot to close it

**Pull request closed without merging**